src/main/java/ch/njol/skript/expressions/ExprIP.java

```diff
@@ -24,97 +24,47 @@
 
 import org.bukkit.entity.Player;
 import org.bukkit.event.Event;
-import org.bukkit.event.player.PlayerLoginEvent;
 import org.eclipse.jdt.annotation.Nullable;
 
-import ch.njol.skript.ScriptLoader;
 import ch.njol.skript.Skript;
 import ch.njol.skript.doc.Description;
 import ch.njol.skript.doc.Examples;
 import ch.njol.skript.doc.Name;
 import ch.njol.skript.doc.Since;
 import ch.njol.skript.expressions.base.SimplePropertyExpression;
-import ch.njol.skript.lang.Expression;
-import ch.njol.skript.lang.ExpressionType;
-import ch.njol.skript.lang.SkriptParser.ParseResult;
-import ch.njol.skript.lang.util.SimpleExpression;
-import ch.njol.util.Kleenean;
 
 @Name("IP")
-@Description("The IP address of a player.")
+@Description("The IP address of player(s).")
 @Examples({"IP-ban the player # is equal to the next line",
 		"ban the IP-address of the player",
 		"broadcast \"Banned the IP %IP of player%\""})
-@Since("1.4, 2.2-dev26 (when used in connect event)")
-public class ExprIP extends SimpleExpression<String> {
+@Since("1.4, 2.2-dev35 (Converted to SimplePropertyExpression)")

```

Comment:
**\* Put a big facepalm image here \***

Who cares about a java thing in Skript? Why you remove the old since (dev26)?

Reply:
Have to define what happened at the version or else you wonder why the version was specified there. Mainly for developers to understand. If it gets generated in the documentation then idk. In dev26 there was the ability to get the address from the connect event, but that was useless in a SimplePropertyExpression and three versions is alot to say.

Reply:
We use Git to trace back changes if needed. Docs are for scripters, mainly.

Reply:
or just don't specify it at all. More or less, nothing changed functionally.